tests/integration/golang/mlflow/run/log_parameter_test.go

```diff
@@ -1,3 +1,190 @@
+	suite.Suite
+	client             *helpers.HttpClient
+	runFixtures        *fixtures.RunFixtures
+	experimentFixtures *fixtures.ExperimentFixtures
+	run                *models.Run
+}
+
+func TestLogParamTestSuite(t *testing.T) {
+	suite.Run(t, new(LogParamTestSuite))
+}
+
+func (s *LogParamTestSuite) SetupTest() {
+	s.client = helpers.NewHttpClient(os.Getenv("SERVICE_BASE_URL"))
+	runFixtures, err := fixtures.NewRunFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.runFixtures = runFixtures
+	expFixtures, err := fixtures.NewExperimentFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.experimentFixtures = expFixtures
+
+	expName := uuid.New().String()
+	exp := &models.Experiment{
+		Name:           expName,
+		LifecycleStage: models.LifecycleStageActive,
+	}
+	_, err = s.experimentFixtures.CreateTestExperiment(context.Background(), exp)
+	assert.Nil(s.T(), err)
+
+	runID := uuid.New().String()
+	run := &models.Run{
+		ID:             strings.ReplaceAll(runID, "-", ""),
+		ExperimentID:   *exp.ID,
+		SourceType:     "JOB",
+		LifecycleStage: models.LifecycleStageActive,
+		Status:         models.StatusRunning,
+	}
+	run, err = s.runFixtures.CreateTestRun(context.Background(), run)
+	assert.Nil(s.T(), err)
+	s.run = run
+}
+
+func (s *LogParamTestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.runFixtures.UnloadFixtures())
+	}()
+
+	req := request.LogParamRequest{
+		RunID: s.run.ID,
+		Key:   "key1",
+		Value: "value1",
+	}
+	resp := map[string]any{}
+	err := s.client.DoPostRequest(
+		fmt.Sprintf("%s%s", mlflow.RunsRoutePrefix, mlflow.RunsLogParameterRoute),
+		req,
+		&resp,
+	)
+	assert.Nil(s.T(), err)
+	assert.Empty(s.T(), resp)
+}
+
+func (s *LogParamTestSuite) TestBatch_Ok() {
+	tests := []struct {
+		name    string
+		request *request.LogBatchRequest
+	}{
+		{
+			name: "Batch of one should succeed",
+			request: &request.LogBatchRequest{
+				RunID: s.run.ID,
+				Params: []request.ParamPartialRequest{
+					{
+						Key:   "key1",
+						Value: "value1",
+					},
+				},
+			},
+		},
+		{
+			name: "Duplicate keys with same value should succeed",
+			request: &request.LogBatchRequest{
+				RunID: s.run.ID,
+				Params: []request.ParamPartialRequest{
+					{
+						Key:   "key2",
+						Value: "value2",
+					},
+					{
+						Key:   "key2",
+						Value: "value2",
+					},
+				},
+			},
+		},
+	}
+	var err error
+	resp := map[string]any{}
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+			err = s.client.DoPostRequest(
+				fmt.Sprintf("%s%s", mlflow.RunsRoutePrefix, mlflow.RunsLogBatchRoute),
+				tt.request,
+				&resp,
+			)
+			assert.Nil(s.T(), err)
+			assert.Empty(s.T(), resp)
+		})
+	}
+}
+
+func (s *LogParamTestSuite) TestBatch_Error() {
+	defer func() {
+		assert.Nil(s.T(), s.runFixtures.UnloadFixtures())
+	}()
+
+	var testData = []struct {
+		name    string
+		error   string
+		request *request.LogBatchRequest
+	}{
+		{
+			name:  "Missing required field",
```

Comment:
the same problem with naming.

Reply:
please add into the name, which required field is missing. it immediately give you all needed information when test failed.

Reply:
I think human readable in the test source file is probably more important than the error spit out by failure

Reply:
`--- PASS: TestLogBatchTestSuite (0.03s)
    --- PASS: TestLogBatchTestSuite/Test_Error (0.03s)
        --- PASS: TestLogBatchTestSuite/Test_Error/Missing_required_field (0.00s)
        --- PASS: TestLogBatchTestSuite/Test_Error/Duplicate_keys_with_different_values_should_error (0.00s)`

Reply:
convention is do not use `whitespaces` because as you can see they will be remove with `underscores` and when test failed, then you will try to fins something like `Missing_required_field` over the code but..... your test have `whitesapces`

Reply:
so the name has to be `MissingRequiredField` + add more details about which filed is missing.